tests/gtest/avifgridapitest.cc

```diff
@@ -4,6 +4,7 @@
 #include <vector>
 
 #include "avif/avif.h"
+#include "avif/internal.h"
```

Comment:
Is there any visibility applied when a shared object is built or are the tests always linking a static lib/objects? If the shared object is used and restricts the visible symbols like it should, this will fail to link.

Reply:
I see `7/25 Test  #8: avifgridapitest ... Passed` in the [build-shared-local (ubuntu-latest)](https://github.com/AOMediaCodec/libavif/actions/runs/4322639324/jobs/7545288861) presubmit so I guess it works. It means the visible symbols are not restricted as they should be? Can this be configured in CMakeLists.txt?

Reply:
I just tested this. When I configure libavif with `cmake -DBUILD_SHARED_LIBS=ON`, two static libraries are produced:  libavif_apps.a  and libavif_internal.a. So avifgridapitest is linked with libavif_internal.a.

The `readelf` command shows `avifImageCopySamples` is not exported from the shared library:

```
$ readelf --dyn-syms libavif.so.15.0.1 | grep avifImageCopy
    61: 00000000000054da   579 FUNC    GLOBAL DEFAULT   12 avifImageCopy
```